src/backend/gl/src/lib.rs

```diff
@@ -66,7 +65,7 @@ impl c::Backend for Backend {
     type Buffer = native::Buffer;
     type UnboundImage = device::UnboundImage;
     type Image = native::Image;
-    type Sampler = native::Sampler;
+    type Sampler = native::FatSampler;
```

Comment:
we could have a sampler as an `enum` to avoid dragging the info with us
I don't think it's a big deal though

Reply:
Implemented, we probably gain nothing from it (size) but it makes the difference a bit clearer.